p2p/net/conn/listen.go

```diff
@@ -20,6 +22,26 @@ import (
 const SecioTag = "/secio/1.0.0"
 const NoEncryptionTag = "/plaintext/1.0.0"
 
+const connAcceptBuffer = 32
+const NegotiateReadTimeout = time.Second * 20
```

Comment:
no, this will _absolutely kill_ users in bad latencies. this is what makes it impossible for users with IP latencies of >20s to use the web of today. ipfs MUST NOT do that. we've been over this many times.